storage/file_adapter.go

```diff
@@ -12,18 +12,18 @@ import (
 
 var logRoot = "/data/logs"
 
-type adapter struct {
+type fileAdapter struct {
 	files map[string]*os.File
 	mutex sync.Mutex
 }
 
-// NewStorageAdapter returns a pointer to a new instance of a file-based storage.Adapter.
-func NewStorageAdapter() (*adapter, error) {
-	return &adapter{files: make(map[string]*os.File)}, nil
+// NewFileAdapter returns an Adapter that uses a file.
+func NewFileAdapter() (Adapter, error) {
```

Comment:
With the `storage.Adapter` impls all re-organized into one flat `storage` package, these init functions are in the same package as the `storage.NewAdapeter(...)` factory function. As such, I think we should no longer export these init functions, as they aren't really (or shouldn't be) part of the `storage` package's public interface.